velero-plugins/pod/restore.go

```diff
@@ -29,6 +33,45 @@ func (p *RestorePlugin) AppliesTo() (velero.ResourceSelector, error) {
 	}, nil
 }
 
+// Check if pod has restore hooks via pod annotations or via restore hook rules
+func (p *RestorePlugin) podHasRestoreHooks(pod corev1API.Pod, resources []velerov1.RestoreResourceHookSpec) (bool, error) {
+	_, postRestoreHookDefined := pod.Annotations[common.PostRestoreHookAnnotation]
+	_, initContainerRestoreHookDefined := pod.Annotations[common.InitContainerRestoreHookAnnotation]
+	if postRestoreHookDefined || initContainerRestoreHookDefined {
+		p.Log.Info("[pod-restore] pod has restore hooks via annotations")
+		return true, nil
+	}
+	p.Log.Info("[pod-restore] pod has no restore hooks via annotations")
+	groupResource := schema.GroupResource{
+		Group:    pod.GroupVersionKind().Group,
+		Resource: pod.GroupVersionKind().Kind,
```

Comment:
Hmm. This doesn't seem quite right. `resource` and `kind` have different values. Since we're only dealing with pods here, you may want to do what Velero does and just refer to the Pods GR:
```
schema.GroupResource{Group: "", Resource: "pods"}
```